### PR TITLE
test: extend integration test TLS capture capabilities.

### PR DESCRIPTION
**Pull request by htuch**
Previously, when setting CAPTURE_PATH, the capture socket extension was
used to generate a proto trace of the post-TLS traffic, suitable for
conversion with capture2pcap to pcap. This patch adds the ability to
capture the pre-TLS traffic via tcpdump. This is useful when debugging
TLS handshakes and other low-level TLS code.

The patch only affects integration tests when CAPTURE_PATH is set, so we
do not introduce a broader system dependency (alternatives considered
were libpcap).

As a bonus, fixed bit rot in CAPTURE_PATH support (s/ssl/tls/) and added
a CI test to verify we are able to capture.

Risk level: Low
Testing: Added a bazel.asan check for basic pcap/pb_text capture.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
Will need https://github.com/envoyproxy/envoy/pull/4850 to pass CI.

**Comment by htuch** [comment]
See Slack thread on #envoy-dev - we need NET_RAW and NET_ADMIN, which isn't available in CircleCI images. Will see what I can do with a machine executor here.

**Comment by htuch** [comment]
After some digging, it seems the resource limits on machine executors and CI slots make adding this support cost prohibitive, so let's just soft fail on tcpdump in CI for now.

File: test/integration/tcp_dump.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  // Derive the port filter expression.
+  std::string port_expr;
+  for (uint32_t port : ports) {
+    if (!port_expr.empty()) {
+      port_expr += " or ";
+    }
+    port_expr += "tcp port " + std::to_string(port);
+  }
+  ENVOY_LOG_MISC(debug, "tcpdumping iface {} to {} with filter \"{}\"", iface, path, port_expr);
+  // Fork a child process. We use explicit fork/wait over popen/pclose to gain
```
Comment (mattklein123) [inline test/integration/tcp_dump.cc]: I'm not that worried given this is test code, but is it safe to fork here given threads, locks, etc.? Does it happen early enough in the test process that we don't need to worry about that?

Comment (htuch) [reply test/integration/tcp_dump.cc]: I think it's safe given we immediately exec, but the whole pthreads+fork business is a bit of a dumpster fire. Is there a better alternative? :)

**Pull request merged**